dask/array/percentile.py

```diff
@@ -232,8 +237,8 @@ def merge_percentiles(finalq, qs, vals, interpolation="lower", Ns=None):
     combined_vals_counts = merge_sorted(*map(zip, vals, counts))
     combined_vals, combined_counts = zip(*combined_vals_counts)
```

Comment:
if vals / counts are cupy arrays here this is quite inefficient. I believe @shwina figured out that we can use an argsort on `vals` to get a gather map to apply against `counts` which keeps things in nice arrays and in vectorized operations.

Reply:
@shwina mentioned that to me earlier today, would be nice to see results for that. This PR won't attempt to do that to keep it as straightforward as possible though.

Reply:
Would suggest raising an issue on this point for now. Agree we should skip it atm

Reply:
👍 will raise an issue

Reply:
A PR is even better. Thanks Ashwin! 😄

Reply:
This is being worked on in PR ( https://github.com/dask/dask/pull/7172 )